// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Send image data
  rpc SendImage (ImageRequest) returns (HelloReply) {}
  // return results
  rpc SendResults (ResultRequest) returns (HelloReply) {}
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message ImageRequest {
  int32 width = 1;
  int32 height = 2;
  bytes overlap_data = 3;
  bytes overlap_blocks = 4;
  string task_id = 5;
  bytes sharing_box = 6;
}

message ResultRequest {
  string task_id = 1;
  bytes results = 2;
}